src/schema/notifications.ts

```diff
@@ -145,6 +145,26 @@ export class ZimbraNotifications {
 		this.getApolloClient().queryManager.broadcastQueries();
 	};
 
+	// Find the actual folder of the shared folder
+	private findSharedItemId = (item: any) => {
+		let sharedFolderId: any;
+		const cachedData = get(this.cache, 'data.data');
+		const allFolders = Object.keys(cachedData).filter(f =>
+			f.includes('Folder:')
+		);
+		const idSplit = item.id.split(':');
+		for (let folderId in allFolders) {
+			if (
+				cachedData[allFolders[folderId]].ownerZimbraId === idSplit[0] &&
+				cachedData[allFolders[folderId]].sharedItemId === idSplit[1]
+			) {
+				sharedFolderId = cachedData[allFolders[folderId]].id;
```

Comment:
Might want to explain break condition in comment.

Reply:
sharedFolderId = allFolders[folderId][id]

Reply:
@itsmeprashant `allFolders `is an array of keys of type `Folder:`, for example - `['Folder:2', 'Folder:1999']`. It is different from cached data.
@avadhutprabhudesai Added comment.